tests/test_data.py

```diff
@@ -31,10 +31,11 @@
     # --------------------------------
     # -----  Test task matching  -----
     # --------------------------------
-    ds_vars = set(ds.samples_paths[0].keys())
     tasks = set(opts.tasks)
     tasks.add("x")
-    assert ds_vars.issubset(tasks)
+    for sample_path in ds.samples_paths:
```

Comment:
I really think we should use `filter_samples`! We don't care if some samples have *more* key, we want them to have *at least* the keys in `tasks` and remove extra-ones

Reply:
oups that's in the test sorry!

Reply:
I thought it was in filter_sampels